docs/build/add-attachments.md

```diff
@@ -0,0 +1,215 @@
+
+The [xmtp.chat](http://xmtp.chat) reference implementation uses Web3 Storage, with XMTP Labs hosting with our own token. We wanted to use a decentralized solution, which made Web3 Storage a good choice. Hosting with our own token reduces user friction, but we cap uploads at 5 MB. In the future, [xmtp.chat](http://xmtp.chat) might give users the option to provide their own token if they want to upload larger file sizes or keep their images beyond a certain timeframe. 
+
+## Security considerations
+
+:::caution
+
+Automatically loading attachments from untrusted users can have negative security implications. Be sure to consider this before enabling attachments to autoload in your app.
+
+:::
+
+A more secure pattern you might want to use is to require the user to click a “Click to load” CTA before loading an attachment. After the attachment has been loaded, it can be pulled from a cache. This pattern requires one more click for the user, but provides greater security.
+
+The [xmtp.chat](http://xmtp.chat) reference implementation intentionally demonstrates both patterns for your reference:
+
+- Less secure: Attachments under 5 MB autoload
+- More secure: Attachments ofer 5 MB require a "Click to load" on first load
+
+## UI considerations
+
+There are a few considerations worth mentioning on the UI side:
+
+- Provide drag-and-drop functionality for users if you’re building a web app.
+- Provide zoom capability for images.
+- Provide a loading state, as attachments can be slower to send than a message with just text. You might choose to optimistically send attachments or display a loading state.
+- Provide fallback text that displays for the recipient if they are using an app that doesn't yet support attachments.
+
+The [xmtp.chat](http://xmtp.chat) reference implementation includes the following error handling states, which you might want to consider for your app as well:
+
+- File size limit exceeded  
+    If your app enforces a file size limit, provide error handling when the size limit is exceeded.
+- Unsupported file type  
+    Provide error handling when a user attempts to attach an unsupported file type.
+- Error sending an attachment  
+    Provide this general error handling to cover other errors when sending an attachment.
+
+To help users avoid these error states in the first place, consider providing UI text that proactively lets users know about file size limits and supported file types.
+
+## Performance considerations
+
+To make your app more performant when loading attachments, consider caching these image URLs after the first load.
+
+The [xmtp.chat](http://xmtp.chat) reference implementation does this using Dexie, but there are other options as well.
+
+Note that when caching attachments, you must reset the cache after the client changes. This is because the client is used to generate the attachment URL that gets added to the cache originally.
```

Comment:
I have to admit that I don't quite understand the last sentence - but perhaps it is b/c I'm not a dev. Any thoughts? =)